common/sampling.h

```diff
@@ -102,3 +102,8 @@ std::string common_sampler_type_to_str(enum common_sampler_type cnstr);
 
 std::vector<enum common_sampler_type> common_sampler_types_from_names(const std::vector<std::string> & names, bool allow_alt_names);
 std::vector<enum common_sampler_type> common_sampler_types_from_chars(const std::string & chars);
+
+#ifdef LLAMA_USE_LLGUIDANCE
+struct llama_sampler * llama_sampler_init_llg(const llama_vocab * vocab,
+        const char * grammar_kind, const char * grammar_data);
+#endif // LLAMA_USE_LLGUIDANCE
```

Comment:
This `ifdef` should be moved inside the implementation. Generally, we want to avoid `ifdef`s in the headers.